CI lints only the default build, so build-tagged files are never checked

`.github/actions/lint/action.yml` runs `golangci-lint-action` with no build tags
and no `GOOS` matrix, so every file behind a build constraint is invisible to
the linter. `make test` does not have this gap — its `TAGS_FLAGS` comes from
`BUILTIN`, which defaults to `all` — so the built-in plugins are tested but
never linted.

## What is currently missed

```console
$ golangci-lint run                                              # what CI runs
0 issues.

$ golangci-lint run --build-tags "builtin_cloudflare builtin_opendht"
14 issues (errcheck)
```

<details>
<summary>The 14 findings</summary>

```
internal/plugin/builtin/cloudflare/cloudflare.go:138:23: Error return value of `resp.Body.Close` is not checked
internal/plugin/builtin/opendht/opendht.go:273:23:      Error return value of `resp.Body.Close` is not checked
internal/plugin/builtin/opendht/opendht_test.go:74,75,76,94,95,96,123,135,136,255,522,564: Error return value of `fmt.Fprintln` is not checked
```

</details>

None of these are urgent — they are `Close` in a `defer` and `Fprintln` to an
`httptest` recorder. The point is that nothing would have caught them, and
nothing will catch the next one.

## Also unguarded, currently clean

Checked while measuring, all reporting 0 issues today, but equally unlinted in
CI:

| Variant | Files it covers |
| --- | --- |
| `--build-tags wgcli` | `internal/wg/client_cli.go` |
| `GOOS=darwin` | `internal/stun/stun_darwinbsd.go`, `internal/ctrl/ping_monitor_darwinbsd.go`, `internal/wg/update_only_darwin.go` |
| `GOOS=freebsd` | `internal/wg/update_only_freebsd.go`, and `client_cli.go` via `freebsd && !wgctrl` |

The darwin/bsd STUN implementation is the part of the codebase with the most
platform-specific logic (pcap handles, BPF offsets that differ per link type),
and it is the part the linter never sees.

## Options

1. Add `--build-tags` to the existing lint step with all built-ins. One line,
   covers the concrete findings above, leaves the `GOOS` gap.
2. Matrix the lint job over `GOOS` × build tags, mirroring what `Build` already
   does. Covers everything, costs several more jobs.
3. `golangci-lint` supports `build-tags` in `.golangci.yml`, which would also
   apply to local runs — worth considering so local and CI agree either way.

Option 1 plus a `GOOS=darwin` run would cover the risky parts without a full
matrix.

Whichever is chosen, the 14 findings need clearing first or the job goes red on
the first run.

--- a/.github/actions/lint/action.yml
+++ b/.github/actions/lint/action.yml
@@ -6,6 +6,10 @@ inputs:
     description: 'Go version to use'
     required: true
     default: 'stable'
+  goos:
+    description: 'GOOS to lint for; golangci-lint only sees the files its GOOS selects'
+    required: false
+    default: 'linux'
 
 runs:
   using: 'composite'
@@ -15,7 +19,11 @@ runs:
         go-version: ${{ inputs.go-version }}
         cache: false
 
+    # Build tags come from .golangci.yaml, so a bare golangci-lint run locally
+    # checks the same files this does.
     - name: golangci-lint
       uses: golangci/golangci-lint-action@v9
       with:
         version: latest
+      env:
+        GOOS: ${{ inputs.goos }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -29,12 +29,16 @@ jobs:
   lint:
     name: Lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [linux, darwin, freebsd]
     steps:
       - uses: actions/checkout@v7
       - name: Lint
         uses: ./.github/actions/lint
         with:
           go-version: ${{ env.GO_VERSION }}
+          goos: ${{ matrix.goos }}
 
   test:
     name: Test

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -12,12 +12,16 @@ jobs:
   lint:
     name: Lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [linux, darwin, freebsd]
     steps:
       - uses: actions/checkout@v7
       - name: Lint
         uses: ./.github/actions/lint
         with:
           go-version: ${{ env.GO_VERSION }}
+          goos: ${{ matrix.goos }}
 
   test:
     name: Test

--- a/.golangci.yaml
+++ b/.golangci.yaml
@@ -1,4 +1,8 @@
 version: "2"
+run:
+  build-tags:
+    - builtin_cloudflare
+    - builtin_opendht
 linters:
   enable:
     - protogetter

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -31,6 +31,18 @@ go test ./internal/entity -v     # Run entity tests
 go test ./internal/repo -v       # Run repository tests
 ```
 
+### Linting
+```bash
+make lint                        # Lint for linux, darwin and freebsd
+make lint LINT_PLATFORMS=linux   # Just the host, for a faster loop
+```
+
+Build tags live in `.golangci.yaml` (`run.build-tags`), not in the Makefile, so
+a bare `golangci-lint run` checks the same files CI does. The `GOOS` sweep
+cannot come from the config, which is why `make lint` loops: golangci-lint only
+sees the files its `GOOS` selects, and `internal/stun/stun_darwinbsd.go` is the
+most platform-specific code in the tree. CI runs the same three as a matrix.
+
 ### Dependency Management
 ```bash
 go mod tidy                      # Clean up dependencies

--- a/Makefile
+++ b/Makefile
@@ -102,6 +102,19 @@ fmt:
 vet:
 	go vet ${TAGS_FLAGS} ./...
 
+# Platforms to lint. golangci-lint only sees the files its GOOS selects, and
+# the darwin/bsd STUN implementation is the most platform-specific code here,
+# so linting the host alone leaves it unchecked. Build tags come from
+# .golangci.yaml so that a bare golangci-lint run agrees with this.
+LINT_PLATFORMS ?= linux darwin freebsd
+
+.PHONY: lint
+lint:
+	@for os in $(LINT_PLATFORMS); do \
+		echo "Linting GOOS=$$os..."; \
+		GOOS=$$os golangci-lint run || exit 1; \
+	done
+
 .PHONY: install
 install: build
 	install -d $(BINDIR)

--- a/internal/plugin/builtin/cloudflare/cloudflare.go
+++ b/internal/plugin/builtin/cloudflare/cloudflare.go
@@ -135,7 +135,7 @@ func (p *CloudflarePlugin) doRequest(ctx context.Context, method, path string, b
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	defer func() { _ = resp.Body.Close() }()
 
 	data, err := io.ReadAll(resp.Body)
 	if err != nil {

--- a/internal/plugin/builtin/opendht/opendht.go
+++ b/internal/plugin/builtin/opendht/opendht.go
@@ -270,7 +270,7 @@ func (p *OpenDHTPlugin) doRequestTo(ctx context.Context, endpoint, method, key s
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	defer func() { _ = resp.Body.Close() }()
 
 	data, err := io.ReadAll(resp.Body)
 	if err != nil {

--- a/internal/plugin/builtin/opendht/opendht_test.go
+++ b/internal/plugin/builtin/opendht/opendht_test.go
@@ -71,9 +71,9 @@ func TestGetReturnsMostRecentValue(t *testing.T) {
 	// 10-minute expiry leaves several of our own under it at once. The proxy
 	// does not order them, so the newest must be selected by ts.
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, line(t, defaultMagic, 200, "newest"))
-		fmt.Fprintln(w, line(t, defaultMagic, 100, "oldest"))
-		fmt.Fprintln(w, line(t, defaultMagic, 150, "middle"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 200, "newest"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 100, "oldest"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 150, "middle"))
 	}))
 	defer server.Close()
 
@@ -91,9 +91,9 @@ func TestGetIgnoresForeignValues(t *testing.T) {
 	// Anyone may publish under a known key. Values that are not our envelope
 	// must not be returned -- not even one whose ts would otherwise win.
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, rawLine(t, "total garbage not json"))
-		fmt.Fprintln(w, line(t, "someone-else", 1<<40, "hijacked"))
-		fmt.Fprintln(w, line(t, defaultMagic, 100, "ours"))
+		_, _ = fmt.Fprintln(w, rawLine(t, "total garbage not json"))
+		_, _ = fmt.Fprintln(w, line(t, "someone-else", 1<<40, "hijacked"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 100, "ours"))
 	}))
 	defer server.Close()
 
@@ -120,7 +120,7 @@ func TestGetEmptyKey(t *testing.T) {
 
 func TestGetOnlyForeignValues(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, line(t, "someone-else", 100, "theirs"))
+		_, _ = fmt.Fprintln(w, line(t, "someone-else", 100, "theirs"))
 	}))
 	defer server.Close()
 
@@ -132,8 +132,8 @@ func TestGetOnlyForeignValues(t *testing.T) {
 
 func TestGetRespectsConfiguredMagic(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, line(t, defaultMagic, 200, "default-magic"))
-		fmt.Fprintln(w, line(t, "custom", 100, "custom-magic"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 200, "default-magic"))
+		_, _ = fmt.Fprintln(w, line(t, "custom", 100, "custom-magic"))
 	}))
 	defer server.Close()
 
@@ -252,7 +252,7 @@ func TestRoundTrip(t *testing.T) {
 			stored = string(body)
 			return
 		}
-		fmt.Fprintln(w, stored)
+		_, _ = fmt.Fprintln(w, stored)
 	}))
 	defer server.Close()
 
@@ -519,7 +519,7 @@ func TestFailsOverToNextEndpoint(t *testing.T) {
 			live := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				liveHits++
 				if r.Method == http.MethodGet {
-					fmt.Fprintln(w, line(t, defaultMagic, 1, "payload"))
+					_, _ = fmt.Fprintln(w, line(t, defaultMagic, 1, "payload"))
 					return
 				}
 				w.WriteHeader(http.StatusOK)
@@ -561,7 +561,7 @@ func TestDoesNotTryLaterEndpointsOnSuccess(t *testing.T) {
 	defer second.Close()
 
 	first := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, line(t, defaultMagic, 1, "payload"))
+		_, _ = fmt.Fprintln(w, line(t, defaultMagic, 1, "payload"))
 	}))
 	defer first.Close()
 

